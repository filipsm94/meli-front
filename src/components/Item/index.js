import React from 'react';
import CurrencyFormat from 'react-currency-format';
import { Link, useRouteMatch } from 'react-router-dom';
import classes from './item.module.scss';

const Item = ({ currency, idProduct, urlImg, amount, title }) => {
    const currency = ` ${props.currency}`;
    let match = useRouteMatch();

    return (
        <Link className={classes.Card} to={{
            pathname: `${match.url}/${props.idProduct}`
        }}>
            <div className={classes.Image}>
                <img src={props.urlImg} alt={props.title} />
            </div>
            <div className={classes.Detail}>
                <div className={classes.Product}>
                    <CurrencyFormat className={classes.Price} value={props.amount} displayType={'text'} thousandSeparator={true} prefix={'$ '} suffix={currency} />
                    <h3>{props.title}</h3>
                </div>
                <div className={classes.Reference}>address</div>
            </div>
        </Link>
    )
}

export default Item;