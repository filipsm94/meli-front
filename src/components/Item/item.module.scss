@import '../../variables';

.Card{
    display: flex;
    align-items: center;
    margin: 1rem 0;
    background-color: $background-white;
    border-radius: 1rem;
    padding: 1rem;
    box-sizing: content-box;
    text-decoration: none;
    color: black;
}

.Image{
    border: 1px solid $background-gray;
    border-radius: 4px;
    padding: 2px;
    width: 15%;
    text-align: center;
    margin-right: 1rem;
}

.Image img{
    width: 150px;
    height: 200px;
}

.Detail{
    width: 85%;
    display: flex;
    align-items: flex-start;
    justify-content: space-between;
}
.Price{
    font-size: xx-large;
}

.Product{
    width: 80%;
}

.Reference{
    width: 20%;
}