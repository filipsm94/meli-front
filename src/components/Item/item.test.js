import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Item from './index';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useRouteMatch: () => ({
    url: 'hola'
  }),
  Link: ({ children, ...rest }) => <a data-testid="link" {...rest} >{children}</a>
}));

describe('Item Component', () => {
  test('renders', () => {
    const props = {
      currency: 'COP',
      idProduct: 'MA124512',
      urlImg: 'url',
      amount: '1000',
      title: 'hola',
    }
    const { container } = render(<Item {...props} />);
    expect(container).toMatchSnapshot();
  });

})
