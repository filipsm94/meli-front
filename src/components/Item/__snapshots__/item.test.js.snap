// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Component renders 1`] = `
<div>
  <a
    class="Card"
    data-testid="link"
    to="[object Object]"
  >
    <div
      class="Image"
    >
      <img
        alt="hola"
        src="url"
      />
    </div>
    <div
      class="Detail"
    >
      <div
        class="Product"
      >
        <span
          class="Price"
        >
          $ 1,000 COP
        </span>
        <h3>
          hola
        </h3>
      </div>
      <div
        class="Reference"
      >
        address
      </div>
    </div>
  </a>
</div>
`;
